exporters/autoexport/metrics.go

```diff
@@ -94,4 +98,43 @@ func init() {
 	RegisterMetricReader("none", func(ctx context.Context) (metric.Reader, error) {
 		return newNoopMetricReader(), nil
 	})
+	RegisterMetricReader("prometheus", func(ctx context.Context) (metric.Reader, error) {
+		host := os.Getenv("OTEL_EXPORTER_PROMETHEUS_HOST")
+		if host == "" {
+			host = "localhost"
+		}
+
+		port := os.Getenv("OTEL_EXPORTER_PROMETHEUS_PORT")
+		if port == "" {
+			port = "9464"
+		}
+
+		go serve(ctx, &http.Server{
+			Addr:         host + ":" + port,
+			ReadTimeout:  5 * time.Second,
+			WriteTimeout: 10 * time.Second,
+			IdleTimeout:  120 * time.Second,
+		})
+
+		return prometheus.New()
```

Comment:
We have a goroutine leak. The HTTP server should be closed when the user closes the returned `metric.Reader`. I think we should create a dedicated implementation of `metric.Reader` to handle shutdown which will take care of cleaning up of both the HTTP server and Prometheus exporter.

Reply:
Agreed. I'll work on that.

Reply:
Done.